interp/constrextern.ml

```diff
@@ -274,17 +274,8 @@ let pattern_printable_in_both_syntax (ind,_ as c) =
 
  (* Better to use extern_glob_constr composed with injection/retraction ?? *)
 let rec extern_cases_pattern_in_scope (scopes:local_scopes) vars pat =
-  (* pboutill: There are letins in pat which is incompatible with notations and
-     not explicit application. *)
-  match pat with
-    | PatCstr(loc,cstrsp,args,na)
-	when !Flags.in_debugger||Inductiveops.constructor_has_local_defs cstrsp ->
-      let c = extern_reference loc Id.Set.empty (ConstructRef cstrsp) in
-      let args = List.map (extern_cases_pattern_in_scope scopes vars) args in
-      CPatCstr (loc, c, Some (add_patt_for_params (fst cstrsp) args), [])
-    | _ ->
   try
-    if !Flags.raw_print || !print_no_symbol then raise No_match;
+    if !Flags.in_debugger || !Flags.raw_print || !print_no_symbol then raise No_match;
```

Comment:
(Sorry github decided for me to reply to the whole PR instead of this thread)

> we should definitely try different recipes for disaster

Do you mean that having independent printers looks also dangerous to you? IMHO what matters is that the debug printers does not lie, and that the behavior of the system does not change when switching to debug mode. That last part is less clear if we catch more exceptions, depending on `is_debugger`.

For instance, it could be really hard to debug a printing time exception (say one due to threading the wrong environment to the printer), don't you think?